router/underlayproviders/udpip/udpip_test.go

```diff
@@ -0,0 +1,294 @@
+// Copyright 2023 ETH Zurich
+// Copyright 2025 SCION Association
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package udpip
+
+import (
+	"crypto/rand"
+	"encoding/binary"
+	"hash/fnv"
+	"net/netip"
+	"testing"
+	"time"
+
+	"github.com/gopacket/gopacket"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/scionproto/scion/pkg/addr"
+	"github.com/scionproto/scion/pkg/private/serrors"
+	"github.com/scionproto/scion/pkg/private/util"
+	"github.com/scionproto/scion/pkg/scrypto"
+	"github.com/scionproto/scion/pkg/slayers"
+	"github.com/scionproto/scion/pkg/slayers/path"
+	"github.com/scionproto/scion/pkg/slayers/path/scion"
+)
+
+var (
+	testKey = []byte("testkey_xxxxxxxx")
+)
+
+func computeMAC(t *testing.T, key []byte, info path.InfoField, hf path.HopField) [path.MacLen]byte {
+	mac, err := scrypto.InitMac(key)
+	require.NoError(t, err)
+	return path.MAC(mac, info, hf, nil)
+}
+
+// Prepares a message that is arriving at its last hop, incoming through interface 1.
+func prepBaseMsg(t *testing.T, payload []byte, flowId uint32) *slayers.SCION {
```

Comment:
`payload` is unused.

```suggestion
func prepBaseMsg(t *testing.T, flowId uint32) *slayers.SCION {
```

Reply:
done. Indeed, not needed. Gone.
<!-- Reviewable comment -OJXzEu_ETKf_72_HKyo:b-m81uon -->
<!-- Sent from Reviewable.io -->